react/components/AccordionFilterContainer.js

```diff
@@ -1,111 +1,101 @@
-import React, { Component } from 'react'
+import React, { useState } from 'react'
 import PropTypes from 'prop-types'
 import { injectIntl, intlShape } from 'react-intl'
 import classNames from 'classnames'
 import { IconCaret } from 'vtex.dreamstore-icons'
 
-import { mountOptions } from '../constants/SearchHelpers'
 import AccordionFilterItem from './AccordionFilterItem'
 
 import searchResult from '../searchResult.css'
 
-class AccordionFilterContainer extends Component {
-  static propTypes = {
-    /** Current available filters */
-    filters: PropTypes.arrayOf(PropTypes.object),
-    /** Intl instance */
-    intl: intlShape,
-    /** Filters mapped for checkbox */
-    filtersChecks: PropTypes.object,
-    /** Checkbox hit callback function */
-    onFilterCheck: PropTypes.func,
-    /** Filters selected previously */
-    selectedFilters: PropTypes.array,
-    isOptionSelected: PropTypes.func.isRequired,
-    map: PropTypes.string.isRequired,
-    rest: PropTypes.string.isRequired,
-  }
-
-  state = {
-    openedItem: null,
-  }
-
-  handleItemSelected = onClose => e => {
-    onClose(e)
+const AccordionFilterContainer = ({
+  filters,
+  intl,
+  onFilterCheck,
+  isOptionSelected,
+}) => {
+  const [openItem, setOpenItem] = useState(null)
 
-    this.setState({
-      openedItem: null,
-    })
-  }
-
-  handleOpen = id => e => {
+  const handleOpen = id => e => {
     e.preventDefault()
 
-    if (this.state.openedItem === id) {
-      this.setState({
-        openedItem: null,
-      })
+    if (openItem === id) {
```

Comment:
I prefer a ternary here, but feel free to leave it this way

Reply:
I will leave it this way..